### ISSUE TITLE
media create_content uploads buffer slice string representation instead of content

### ISSUE THREAD

**Issue by stoically** [ruma/ruma-client-api#81]
Given the following code:

```rust
let file = std::fs::read("some.file").unwrap();
client
    .request(r0::media::create_content::Request {
        filename: Some("name".to_owned()),
        content_type: "type".to_owned(),
        file,
    });
```

What ends up on the homeserver in the file is `[123,34,111,98,106...`, resulting in 2MB files being 9MB files instead.

**Comment by jplatte** [comment]
This might actually need changes to `ruma_api!` because I think currently we always serialize the body through serde_json; in this case we need to write it without any serialization.

**Comment by stoically** [comment]
That might be a problem for the media response as well then, if it tries to deserialize non-serialized content.

**Comment by jplatte** [comment]
Yeah, same problem there.

**Comment by jplatte** [comment]
Should be fixed by 258fbf5a0154602612ed6bb18821b62791ff3f7a.
